src/install/extract_tarball.zig

```diff
@@ -287,5 +300,55 @@ fn extract(this: *const ExtractTarball, tgz_bytes: []const u8) !string {
         ) catch break :create_index;
     }
 
+    switch (this.resolution.tag) {
+        .github => {
+            const manager = this.package_manager;
+            var package = manager.lockfile.packages.get(this.package_id);
```

Comment:
packages array isn't thread-safe, this will crash if packages array is resized while extracting

This step will need to happen on the main thread. OR, it could read the package.json file in this other thread and send it off to the main thread to run Lockfile.Package.parse

Reply:
Done in https://github.com/oven-sh/bun/pull/1875/commits/54a00bf39f81f86970914306550e9321369ff525